components/layout_2020/fragment_tree/fragment.rs

```diff
@@ -279,15 +280,15 @@ impl CollapsedBlockMargins {
 impl CollapsedMargin {
     pub fn zero() -> Self {
         Self {
-            max_positive: Length::zero(),
-            min_negative: Length::zero(),
+            max_positive: Au::zero(),
+            min_negative: Au::zero(),
         }
     }
 
-    pub fn new(margin: Length) -> Self {
+    pub fn new(margin: Au) -> Self {
         Self {
-            max_positive: margin.max(Length::zero()),
-            min_negative: margin.min(Length::zero()),
+            max_positive: std::cmp::max(margin, Au::zero()),
+            min_negative: std::cmp::min(margin, Au::zero()),
```

Comment:
No need to switch to `std::cmp::max`.

```suggestion
            max_positive: margin.max(Au::zero()),
            min_negative: margin.max(Au::zero()),
```